common/serial/port_io.h

```diff
@@ -0,0 +1,43 @@
+#ifndef PORT_IO__H
+#define PORT_IO__H
+
+#include <arch.h>
+
+class port_io {
+
+public:
+	port_io(arch_handler *arch) : m_arch(arch) 
+	{
+
+	}
+
+	~port_io(void) 
+	{
+		delete m_arch;
+	}
+
+	void port_write(uint16_t port, uint8_t value)
+	{
+		m_arch->port_write(port, value);
```

Comment:
I think that is a hell of a lot easier to read than GCC's ****ed up inline assembly, and once you add all of the parameters (which be on a line of their own), your going to have basically the same thing.